tst/unit/test_metadata.cpp

```diff
@@ -225,5 +225,12 @@ TEST_CASE("Refinement Information in Metadata", "[Metadata]") {
         REQUIRE_THROWS_AS(m.GetRefinementFunctions(), std::runtime_error);
       }
     }
+    WHEN("We set the refinement flag after the fact") {
+      m.Set(Metadata::FillGhost);
+      THEN("The refinement funcs should be present with a valid label") {
+        const auto &my_funcs = m.GetRefinementFunctions();
+        REQUIRE(my_funcs.label().size() > 0);
+      }
+    }
```

Comment:
Check that the fail-case downstream works. Note that if we apply the change I propose in #844 , this test becomes unnecessary/impossible again.